### ISSUE TITLE
Protect data against unintended modification/deletion

### ISSUE THREAD

**Issue by sylvainloiseau** [MacPass/MacPass#1138]
This may seem unnecessary, but in my experience of using this app (for years now), I'm always feeling anxious of un-intentionaly *editing*/erasing the data (and notably the password) while opening an item in order to *consult* it. And indeed it happen recently that I loose a password through some editing (it was on a backup fortunately).

Since passwords are intended to be read frequently (and created only once), and since they are precious item, I'm wondering whether the detailed info about an entry ("the right panel") couldn't be in read only mode unless the user select a "edit the item" button or something like that.

Hoping it will sound a useful suggestion to others,
Best regards and thanks for this app,
Sylvain Loiseau

**Comment by mstarke** [comment]
This behaviour is on the "drawing board" for ages (7 years 😲 - see #47) and was planed to be changed for more than one release but it always got pushed out. With the current change to incorporate TOTP support, there are some changes to KeePassKit that might make the change to edit/view a bit better.

I wanted to use this to change the Inspector to show only field that have content to make it more compact.

The biggest issue I have to make this work is that if you have an "edit" button you then should have an "save" and a "discard" button. And implementing a discardable edit working in the current setup is rather hard. I might just go for "edit" and "done" and hope for all the user wanting to "cancel" to be able to use undo…

Duplicate of #47.

**Comment by sylvainloiseau** [comment]
Thanks for your answer. It will be a great feature to have theses "edit" and "done" button. Even the "done" button is may be unnecessary (and indeed, if there is no "discard" button, it seems more clear that there be no "done" button either): why not give non-editable fields at first to the user, and an "edit" button that turn the Pane into its current behavior (with editable fields, implicitly saved).
